src/loading.jl

```diff
@@ -0,0 +1,91 @@
+loadleaf!(dst, src, err) = dst
+function loadleaf!(dst::AbstractArray, src::Bool, err)
+  if iszero(src)
+    dst .= src
+  else
+    error("Cannot copy boolean parameter == true to non-zero parameter.")
+  end
+  return dst
+end
+loadleaf!(dst::Bool, src::AbstractArray, err) = iszero(dst) ? dst :
+  error("Cannot copy non-zero parameter to boolean parameter == true.")
+function loadleaf!(dst::AbstractArray, src::AbstractArray, err)
+  (size(dst) == size(src)) || throw(err)
+  copyto!(dst, src)
+end
+
+_tie_check(dst::Bool, src::AbstractArray) = iszero(dst) ||
+  error("Encountered tied parameter with boolean source at some nodes and non-boolean sources at others.")
+_tie_check(dst::AbstractArray, src::Bool) = (iszero(dst) && iszero(src)) ||
+  error("Encountered tied parameter with boolean source at some nodes and non-boolean sources at others.")
+_tie_check(dst::AbstractArray, src::AbstractArray) = (dst == src) ||
+  error("Encountered tied destination parameters with untied and mismatched sources.")
+_tie_check(dst, src) = true
+
+_bool_tie_check(dst, src) = true
+
+"""
+    loadmodel!(dst, src)
+
+Copy all the parameters (trainable and non-trainable) from `src` into `dst`.
+
+Recursively walks `dst` and `src` together using [`Functors.children`](@ref),
+and calling `copyto!` on parameter arrays.
+Non-array elements (such as activation functions) are not copied
+and do not need to match between `dst` and `src`.
+Inactive parameters, encoded by `false` in place on an array,
+can be copied to and from all-zero arrays.
+Attempting to copy a non-zero array to/from an inactive parameter will throw an error.
```

Comment:
Maybe this is more clear?
```suggestion
Inactive parameters can be encoded by using the boolean value `false` instead of an array.
If `src` or `dst` has `false` where the other model has an all-zero array, no error will be raised (and no values copied). However, attempting to copy a non-zero array to/from an inactive parameter will throw an error.
```

Reply:
Okay I used this with a few modifications, because your comment made me realize that the behavior is not 1-1 like the docstring implies.

Reply:
Can we shorten this somehow? It seems a very tiny edge case about layers with `bias=false` vs. models with an actual bias array. Most likely this will never happen in real life. Yet somehow it gets an essay describing all possible paths... 

How about just "Zero bias and `bias=false` are considered equivalent."?

Reply:
Could put it under extended help? See point 11 of https://docs.julialang.org/en/v1/manual/documentation/.

IMO magic should at least be documented...